build/room_ensurer/room_ensurer.py

```diff
@@ -131,88 +124,112 @@ def ensure_rooms(self) -> None:
+                offline_servers=[
+                    server_name
+                    for server_name, room_id in server_name_to_room_ids.items()
+                    if room_id is None
+                ],
+            )
+
+        if not are_all_available_rooms_the_same:
             log.warning(
                 "Room id mismatch",
                 alias_prefix=room_alias_prefix,
-                expected=first_server_room_info.room_id,
+                expected=expected_room_id,
                 found={
-                    server_name: room_info.room_id if room_info else None
-                    for server_name, room_info in room_infos.items()
+                    server_name: room_id
+                    for server_name, room_id in server_name_to_room_ids.items()
+                    if room_id is not None
                 },
             )
-            own_server_room_info = room_infos.get(self._own_server_name)
-            own_server_room_alias = f"#{room_alias_prefix}:{self._own_server_name}"
-            first_server_room_alias = f"#{room_alias_prefix}:{self._first_server_name}"
-            if not own_server_room_info:
-                log.warning(
-                    "Room missing on own server, adding alias",
-                    server_name=self._own_server_name,
-                    room_id=first_server_room_info.room_id,
-                    new_room_alias=own_server_room_alias,
-                )
-                self._join_and_alias_room(first_server_room_alias, own_server_room_alias)
-                log.info("Room alias set", alias=own_server_room_alias)
-            elif own_server_room_info.room_id != first_server_room_info.room_id:
-                log.warning(
-                    "Conflicting local room, reassigning alias",
-                    server_name=self._own_server_name,
-                    expected_room_id=first_server_room_info.room_id,
-                    current_room_id=own_server_room_info.room_id,
-                )
-                self._own_api.remove_room_alias(own_server_room_alias)
-                self._join_and_alias_room(first_server_room_alias, own_server_room_alias)
-                log.info(
-                    "Room alias updated",
-                    alias=own_server_room_alias,
-                    room_id=first_server_room_info.room_id,
-                )
-            else:
-                log.warning("Mismatching rooms on other servers. Doing nothing.")
-        else:
+
+        if not has_unavailable_rooms and are_all_available_rooms_the_same:
             log.info(
-                "Room state ok.",
-                network=network,
-                server_rooms={
-                    server_name: room_info.room_id if room_info else None
-                    for server_name, room_info in room_infos.items()
-                },
+                "Room state ok.", server_rooms=server_name_to_room_ids,
+            )
+
```

Comment:
Isn't a `return` missing here?

Reply:
It needs to continue because of `_ensure_admin_power_levels` in line 233

Reply:
Ah, ok